go-controller/pkg/node/iptables/iptables_manager.go

```diff
@@ -208,91 +224,66 @@ func getChainRuleArgs(ipt iptables.Interface, table iptables.Table, chain iptabl
 				Args: strings.Split(line, " ")[2:],
 			})
 		}
-
 	}
 	return rules, nil
 }
 
-func processRules(ipt iptables.Interface, ownedChains []Chain, wantedChainRules map[Chain][]RuleArg, existingChainRules map[Chain][]RuleArg) error {
+func processRules(ipt iptables.Interface, wantedChainRules map[rulesIndex]rules, existingChainRules map[rulesIndex]rules) error {
 	var err error
-	for _, wantedChain := range ownedChains {
-		err = execIPTablesWithRetry(func() error {
-			_, err = ipt.EnsureChain(wantedChain.Table, wantedChain.Chain)
-			if err != nil {
-				return fmt.Errorf("failed to ensure chain %s in table %s: %v", wantedChain.Chain, wantedChain.Table, err)
-			}
-			return nil
-		})
-		if err != nil {
-			return fmt.Errorf("failed to process iptables rules in table %s: %v", wantedChain.Table, err)
-		}
-	}
-	for existingChain, existingRuleArgs := range existingChainRules {
-		if existingChain.Proto != ipt.Protocol() {
+	// cleanup any rules we see that we do not expect if the chain is 'owned'
+	for rulesIndex, existingRules := range existingChainRules {
+		if rulesIndex.Proto != ipt.Protocol() {
 			continue
 		}
-		wantedRuleArgs, found := wantedChainRules[existingChain]
-		if !found {
-			// we don't delete chains we don't manage, but we ensure rules are removed
-			continue
-		}
-		err = execIPTablesWithRetry(func() error {
-			_, err = ipt.EnsureChain(existingChain.Table, existingChain.Chain)
-			if err != nil {
-				return fmt.Errorf("failed to ensure chain %s in table %s: %v", existingChain.Chain, existingChain.Table, err)
-			}
-			return nil
-		})
-		if err != nil {
-			klog.Errorf("IPTables manager: failed to process iptables rules: %v", err)
-		}
-		var isOwnedChain bool
-		for _, ownedChain := range ownedChains {
-			if ownedChain == existingChain {
-				isOwnedChain = true
-				break
-			}
-		}
+		wantedRules, found := wantedChainRules[rulesIndex]
 		// owned chains means we exclusively control all rules in the chain and therefore remove any rules which we do not want
```

Comment:
oh here is the comment, might be nice to put that in the struct where exclusive is